File: java/src/test/java/io/github/securekit/CsrfTokenManagerTest.java
package io.github.securekit;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class CsrfTokenManagerTest {

    @Test
    void roundTrip() {
        CsrfTokenManager mgr = new CsrfTokenManager(SecureRandomUtil.bytes(32), 3600_000L);
        String token = mgr.generate("session-123");
        assertTrue(mgr.verify("session-123", token));
    }

    @Test
    void rejectsWrongSession() {
        CsrfTokenManager mgr = new CsrfTokenManager(SecureRandomUtil.bytes(32), 3600_000L);
        String token = mgr.generate("session-A");
        assertFalse(mgr.verify("session-B", token));
    }

    @Test
    void rejectsExpiredToken() throws InterruptedException {
        CsrfTokenManager mgr = new CsrfTokenManager(SecureRandomUtil.bytes(32), 5L);
        String token = mgr.generate("session-123");
        Thread.sleep(30);
        assertFalse(mgr.verify("session-123", token));
    }

    @Test
    void rejectsForgedToken() {
        CsrfTokenManager mgr = new CsrfTokenManager(SecureRandomUtil.bytes(32), 3600_000L);
        assertFalse(mgr.verify("session-123", "forged.token.value"));
    }
}

File: php/tests/CsrfTokenManagerTest.php
<?php

declare(strict_types=1);

namespace SecureKit\Tests;

use PHPUnit\Framework\TestCase;
use SecureKit\CsrfTokenManager;
use SecureKit\SecureRandom;

final class CsrfTokenManagerTest extends TestCase
{
    public function testRoundTrip(): void
    {
        $mgr = new CsrfTokenManager(SecureRandom::bytes(32), 3600);
        $token = $mgr->generate('session-123');
        $this->assertTrue($mgr->verify('session-123', $token));
    }

    public function testRejectsWrongSession(): void
    {
        $mgr = new CsrfTokenManager(SecureRandom::bytes(32), 3600);
        $token = $mgr->generate('session-A');
        $this->assertFalse($mgr->verify('session-B', $token));
    }

    public function testRejectsExpiredToken(): void
    {
        $mgr = new CsrfTokenManager(SecureRandom::bytes(32), 1);
        $token = $mgr->generate('session-123');
        usleep(1_100_000);
        $this->assertFalse($mgr->verify('session-123', $token));
    }

    public function testRejectsForgedToken(): void
    {
        $mgr = new CsrfTokenManager(SecureRandom::bytes(32), 3600);
        $this->assertFalse($mgr->verify('session-123', 'forged.token.value'));
    }
}
